packages/nx/src/plugins/js/lock-file/pnpm-parser.ts

```diff
@@ -159,6 +163,13 @@ export function stringifyPnpmLockfile(
   return stringifyToPnpmYaml(output);
 }
 
+function normalizeLockfileVersion(version: string | number) {
+  if (typeof version === 'string' || version < 6) {
```

Comment:
Should this be `typeof version === 'string' || version >= 6`? If using lockfile v5 and version could be a string or number, this function could still return a string or number.

Perhaps adding a strict return type here to `: string` would be useful.